sklearnex/linear_model/linear.py

```diff
@@ -179,10 +182,14 @@ def _onedal_fit_supported(self, method_name, *data):
                     "Forced positive coefficients are not supported.",
                 ),
                 (
-                    not is_underdetermined,
+                    not (is_underdetermined and (not supports_all_variants or is_gpu)),

```

Comment:
```suggestion
                    not is_underdetermined or (supports_all_variants and not is_gpu)),

```
I was getting confused by the logic of the double negations.